docs/development/benchmarking.md

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_position: 3
+---
+
+# Benchmarking
+
+Do you have an existing Redis environment and would like to see if Dragonfly could be a better
+replacement? <br/>
+Are you developing a service and would like to determine which cloud instance type to
+allocate for Dragonfly? <br/>
+Do you wonder how many replicas you need to support your workload?
+
+If so, read on, because this page is for you!
+
+## Choosing an Environment
+
+A benchmark is done to assess the performance aspects of a system. In the case of Dragonfly, a
+benchmark is commonly used to assess the CPU and memory performance & utilization.
+
+Depending on the goals of your benchmark, you should choose the machine size accordingly. For a
+production mimicking benchmark, you should use a machine size and traffic load similar to that of
+your busiest production timing, or even higher to allow for some cushion.
+
+If you do not use a cloud instance, it might be a good idea to configure your CPU's governance to
+performance by issuing:
+
+```shell
+sudo apt install linux-tools-common linux-tools-generic
+sudo cpupower frequency-set --governor performance
+```
+
+Then, when you're done with the benchmark you could reboot your machine or run the following:
+
+```shell
+sudo cpupower frequency-set --governor powersave
+```
+
+## Setting Up Dragonfly
+
+Dragonfly can run in [Docker](/getting-started/docker) or directly installed as a
+[binary](/getting-started/binary) on your machine. See the [Getting Started](/getting-started) page
+for other options and the latest documentation.
+
+## Reducing Noise
+
+Ideally, a benchmark should be run in as similar as possible environment as the production setup.
+
+In busy production deployments, it is common to run Dragonfly in its own machine (virtual or
+dedicated). If you plan to do so in your production setup as well (which we highly recommend),
+consider running the benchmark in a similar way.
+
+In practice, it means that any other systems in your setup (like other services & databases) should
+run in other machines. Importantly, also the software that sends the traffic should run in another
```

Comment:
i would even make the last sentence bold or move it to the start because it's the most relevant sentence here for practical purposes.